app/routes/editor/problems/$problemId.tsx

```diff
@@ -55,18 +55,37 @@ export const action: ActionFunction = async ({ request, params }) => {
 }`;
             break;
         case 'post':
-            const solution = data.get('solution');
-            variables["input"] = {
-                problem: data.get('problem'),
-                solution: solution === '' ? null : solution,
-                type: data.get('type'),
-                tagIds: (data.get('tags') as string).split(',').map(id => parseInt(id, 10))
-            };
-            query = `mutation ProblemUpdate($id: ID!, $input: ProblemInput) {
+            switch (data.get('act')) {
+                case 'update':
+                    const solution = data.get('solution');
+                    variables["input"] = {
+                        problem: data.get('problem'),
+                        solution: solution === '' ? null : solution,
+                        type: data.get('type'),
+                        tagIds: (data.get('tags') as string).split(',').map(id => parseInt(id, 10))
+                    };
+                    query = `mutation ProblemUpdate($id: ID!, $input: ProblemInput) {
     updateProblem(id: $id, input: $input) {
         id
     }
 }`;
+                    break;
+                case 'add':
+                    variables["input"] = {
+                        problem: 'New problem',
+                        solution: null,
+                        type: ProblemType.STATIC,
+                        tagIds: []
+                    };
+                    query = `mutation ProblemAdd($input: ProblemInput) {
+    addProblem(input: $input) {
+        id
+    }
+}`;
```

Comment:
Should we consider not adding new problem immediately, but setting `problem` and other properties first and then saving new problem?